### ISSUE TITLE
Meta query: null; Error while running ConnectionCallable

### ISSUE THREAD

**Issue by ThomasVallet** [jOOQ/jOOQ#9564]
###
Hi,
I'm having an issue with Jooq using meta queries,
I'm doing something like this :
```
Table<?> tableFound = this.dslContext.meta()
                .getCatalog(this.runtimeConfigurationService.getDbCatalogName())
                .getSchema(this.getSchemaName(executionId))
                `.getTable(extractedTable);
```
And I got a warning : 
`SQL exception : Exception while executing meta query: null`
And after that I got an error : 
`org.jooq.exception.DataAccessException: Error while running ConnectionCallable`.
It works sometimes, but sometimes it fails, I don't really understand what's happening. I got this issue on SQL Server, it seems to work well on MySQL.
I don't know if it comes from Sql Server configuration, from Jooq or from my use of meta datas. It fails only when I use `getTable()` in meta datas.
Thanks in advance for your attention.

### Versions:

- jOOQ: 3.11.11
- Java: 11
- Database (include vendor): Mysql 5.1.34 (working well) ; Sql Server 7.4.1.jre11 (failing)
- OS: Windows 10
- JDBC Driver (include name if inofficial driver): ojdbc 12.1.0.1

**Comment by lukaseder** [comment]
Thank you very much for your request.

This doesn't ring a bell, but I suspect you won't be able to run the exact same code on MySQL and SQL Server, because only SQL Server supports catalogs *and* schemas, whereas MySQL only supports catalogs (although jOOQ treats them as schemas).

Would you mind pasting your complete stack trace that shows all the parts that are involved with the error you're getting? Also, what values are you passing to `getCatalog()` and `getSchema()`, in each case?

**Comment by ThomasVallet** [comment]
Hi Lukas,

I tried `getCatalog()` on MySQL and SQLServer, and it does return an empty string on MySQL so it's perfect for me, I can use the exact same code on both databases. It works sometimes.
Unfortunately I can't send you a better stack, I changed the way I get my data. But I had a configuration file where I put my catalog and my schema name. So I got them like that directly with the string name.
It's working for me now, but I still don't get why It worked like sometimes.
